### ISSUE TITLE
shadow - `Ctrl+[` shortcut is not passed

### ISSUE THREAD

**Issue by stdedos** [Xpra-org/xpra#3417]
**Describe the bug**
`Ctrl+[` shortcut is not passed on to the server
![image](https://user-images.githubusercontent.com/133706/148647959-cd86e645-3b7b-4461-bbd2-4da133e1ec20.png)

```py
19,797 New unix-domain connection received
19,797  on '/run/user/1000/xpra/0/socket'
19,799 enabled debugging for:
19,799  - Logger(xpra.x11.x11_server_core, x11, server, keyboard)
19,799  - Logger(xpra.server.rfb.rfb_server, rfb, keyboard)
19,799  - Logger(importlib._bootstrap, x11, bindings, keyboard)
19,799  - Logger(xpra.x11.server_keyboard_config, keyboard)
19,799  - Logger(xpra.server.source.input_mixin, keyboard)
19,799  - Logger(xpra.gtk_common.keymap, keyboard)
19,799  - Logger(xpra.server.mixins.input_server, keyboard)
19,799  - Logger(xpra.x11.xkbhelper, x11, keyboard)
20,001 New unix-domain connection received
20,001  on '/run/user/1000/xpra/0/socket'
20,021 client   1 @32.688 enabled debugging for:
20,059 client   1 @32.688  - Logger(xpra.platform.win32.keyboard, keyboard)
20,059 client   1 @32.688  - Logger(xpra.client.gtk_base.gtk_client_window_base, keyboard, events)
20,059 client   1 @32.688  - Logger(xpra.platform.win32.gui, win32, keyboard)
20,059 client   1 @32.688  - Logger(xpra.client.keyboard_helper, keyboard)
20,059 client   1 @32.688  - Logger(xpra.client.ui_client_base, client, keyboard)
20,059 client   1 @32.688  - Logger(xpra.client.keyboard_shortcuts_parser, keyboard)
20,059 client   1 @32.688  - Logger(xpra.client.gtk_base.gtk_client_window_base, keyboard)
20,059 client   1 @32.688  - Logger(xpra.gtk_common.keymap, keyboard)
20,060 client   1 @32.688  - Logger(xpra.client.client_window_base, keyboard)
20,060 client   1 @32.688  - Logger(xpra.platform.keyboard_base, keyboard)
21,386 client   1 @34.063 mask_to_names(<flags GDK_CONTROL_MASK of type Gdk.ModifierType>) GetKeyState(VK_NUMLOCK)=1, names=['control', 'mod2']
21,386 client   1 @34.063 parse_key_event(<Gdk.EventKey object at 0x0000022d485f3680 (void at 0x0000022d48b17000)>, True)=<xpra.common.KeyEvent object at 0x0000022d48640e20>
21,388 client   1 @34.063 handle_key_action(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), <xpra.common.KeyEvent object at 0x0000022d48640e20>) wid=1
21,388 client   1 @34.063 key_handled_as_shortcut(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), 'Control_R', ['control', 'mod2'], True) shortcuts_enabled=True, shortcuts=None
21,388 client   1 @34.063 send_delayed_key() delayed_event=None
21,388 client   1 @34.063 send_key_action(1, <xpra.common.KeyEvent object at 0x0000022d48640e20>)
21,389 will try levels: [0, 4, 1, 5, 2, 6, 3, 7]
21,389 do_get_keycode(163, 'Control_R', True, ['control', 'mod2'], 65508, '', 0)=105 (level=0, shift=False, mode=0, keysyms=['Control_R'])
21,389 not toggling any modifiers state for keysyms=['Control_R']
21,389 process_key_action(['key-action', 1, 'Control_R', True, ('control', 'mod2'), 65508, '', 163, 0]) server keycode=105, group=0
21,389 set_keyboard_layout_group(0) ignored, value unchanged
21,389 filtered_modifiers_set(['mod2'])={'mod2'}
21,389 modifier 'control' ignored (in ignored keynames={'Control_R', 'Control_L'})
21,389 filtered_modifiers_set(['control', 'mod2'])={'mod2'}
21,389 handle_key((1, True, 'Control_R', 65508, 105, ['control', 'mod2'], True, True))
21,389 handle keycode pressing   105: key 'Control_R'
21,390 fake_key(105, True)
21,507 client   1 @34.172 mask_to_names(<flags GDK_CONTROL_MASK of type Gdk.ModifierType>) GetKeyState(VK_NUMLOCK)=1, names=['control', 'mod2']
21,507 client   1 @34.172 parse_key_event(<Gdk.EventKey object at 0x0000022d485f3680 (void at 0x0000022d48b17000)>, True)=<xpra.common.KeyEvent object at 0x0000022d49cf50a0>
21,508 client   1 @34.172 handle_key_action(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), <xpra.common.KeyEvent object at 0x0000022d49cf50a0>) wid=1
21,508 client   1 @34.172 key_handled_as_shortcut(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), 'U+001B', ['control', 'mod2'], True) shortcuts_enabled=True, shortcuts=None
21,509 client   1 @34.172 send_delayed_key() delayed_event=None
21,509 client   1 @34.172 send_key_action(1, <xpra.common.KeyEvent object at 0x0000022d49cf50a0>)
21,510 will try levels: [0, 4, 1, 5, 2, 6, 3, 7]
21,510 do_get_keycode(219, 'U+001B', True, ['control', 'mod2'], 16777243, '\x1b', 0)=-1, 0 (keyname translation)
21,510 process_key_action(['key-action', 1, 'U+001B', True, ('control', 'mod2'), 16777243, '\x1b', 219, 0]) server keycode=-1, group=0
21,510 filtered_modifiers_set(['control', 'mod2'])={'mod2', 'control'}
21,510 filtered_modifiers_set(['control', 'mod2'])={'mod2', 'control'}
21,614 client   1 @34.281 mask_to_names(<flags 0 of type Gdk.ModifierType>) GetKeyState(VK_NUMLOCK)=1, names=['mod2']
21,614 client   1 @34.281 parse_key_event(<Gdk.EventKey object at 0x0000022d485f32c0 (void at 0x0000022d48b17160)>, False)=<xpra.common.KeyEvent object at 0x0000022d48640e20>
21,614 client   1 @34.281 handle_key_action(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV422P)), <xpra.common.KeyEvent object at 0x0000022d48640e20>) wid=1
21,615 client   1 @34.281 key_handled_as_shortcut(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV422P)), 'Control_R', ['mod2'], False) shortcuts_enabled=True, shortcuts=None
21,615 client   1 @34.281 send_delayed_key() delayed_event=None
21,616 client   1 @34.281 send_key_action(1, <xpra.common.KeyEvent object at 0x0000022d48640e20>)
21,617 process_key_action(['key-action', 1, 'Control_R', False, ('mod2',), 65508, '', 163, 0]) server keycode=105, group=0
21,617 set_keyboard_layout_group(0) ignored, value unchanged
21,617 modifier 'control' ignored (in ignored keynames={'Control_R', 'Control_L'})
21,617 filtered_modifiers_set(['control', 'mod2'])={'mod2'}
21,617 filtered_modifiers_set(['mod2'])={'mod2'}
21,618 handle_key((1, False, 'Control_R', 65508, 105, ['mod2'], True, True))
21,618 handle keycode unpressing 105: key 'Control_R'
21,618 fake_key(105, False)
21,668 client   1 @34.297 mask_to_names(<flags 0 of type Gdk.ModifierType>) GetKeyState(VK_NUMLOCK)=1, names=['mod2']
21,668 client   1 @34.297 parse_key_event(<Gdk.EventKey object at 0x0000022d485f3a90 (void at 0x0000022d48b17000)>, False)=<xpra.common.KeyEvent object at 0x0000022d48640ee0>
21,668 client   1 @34.297 handle_key_action(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV420P)), <xpra.common.KeyEvent object at 0x0000022d48640ee0>) wid=1
21,668 client   1 @34.297 key_handled_as_shortcut(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV420P)), 'bracketleft', ['mod2'], False) shortcuts_enabled=True, shortcuts=None
21,669 client   1 @34.297 send_delayed_key() delayed_event=None
21,669 client   1 @34.297 send_key_action(1, <xpra.common.KeyEvent object at 0x0000022d48640ee0>)
21,669 process_key_action(['key-action', 1, 'bracketleft', False, ('mod2',), 91, '[', 219, 0]) server keycode=34, group=0
21,669 set_keyboard_layout_group(0) ignored, value unchanged
21,669 filtered_modifiers_set(['mod2'])={'mod2'}
21,670 filtered_modifiers_set(['mod2'])={'mod2'}
21,670 is_modifier(34) not found
21,670 handle_key((1, False, 'bracketleft', 91, 34, ['mod2'], False, True))
21,670 handle keycode 34: key bracketleft was already unpressed, ignoring
23,237 client   1 @35.906 mask_to_names(<flags GDK_CONTROL_MASK of type Gdk.ModifierType>) GetKeyState(VK_NUMLOCK)=1, names=['control', 'mod2']
23,238 client   1 @35.906 parse_key_event(<Gdk.EventKey object at 0x0000022d4a899c20 (void at 0x0000022d48b17000)>, True)=<xpra.common.KeyEvent object at 0x0000022d48640ee0>
23,239 client   1 @35.906 handle_key_action(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), <xpra.common.KeyEvent object at 0x0000022d48640ee0>) wid=1
23,239 client   1 @35.906 key_handled_as_shortcut(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), 'Control_L', ['control', 'mod2'], True) shortcuts_enabled=True, shortcuts=None
23,239 client   1 @35.906 process_key_event: Control_L pressed=True, with GetKeyState(VK_RMENU)=1
23,291 client   1 @35.969 send_delayed_key() delayed_event=(<bound method KeyboardHelper.send_key_action of <xpra.client.gtk_base.gtk_keyboard_helper.GTKKeyboardHelper object at 0x0000022d68823460>>, 1, <xpra.common.KeyEvent object at 0x0000022d48640ee0>)
23,292 client   1 @35.969 send_delayed_key() GetKeyState(VK_RMENU)=1
23,293 client   1 @35.969 send_key_action(1, <xpra.common.KeyEvent object at 0x0000022d48640ee0>)
23,293 will try levels: [0, 4, 1, 5, 2, 6, 3, 7]
23,293 do_get_keycode(17, 'Control_L', True, ['control', 'mod2'], 65507, '', 0)=37 (level=0, shift=False, mode=0, keysyms=['Control_L'])
23,294 not toggling any modifiers state for keysyms=['Control_L']
23,294 process_key_action(['key-action', 1, 'Control_L', True, ('control', 'mod2'), 65507, '', 17, 0]) server keycode=37, group=0
23,294 set_keyboard_layout_group(0) ignored, value unchanged
23,294 filtered_modifiers_set(['mod2'])={'mod2'}
23,294 modifier 'control' ignored (in ignored keynames={'Control_R', 'Control_L'})
23,294 filtered_modifiers_set(['control', 'mod2'])={'mod2'}
23,294 handle_key((1, True, 'Control_L', 65507, 37, ['control', 'mod2'], True, True))
23,294 handle keycode pressing    37: key 'Control_L'
23,294 fake_key(37, True)
23,383 client   1 @36.047 mask_to_names(<flags GDK_CONTROL_MASK of type Gdk.ModifierType>) GetKeyState(VK_NUMLOCK)=1, names=['control', 'mod2']
23,383 client   1 @36.047 parse_key_event(<Gdk.EventKey object at 0x0000022d4847fdb0 (void at 0x0000022d48b17160)>, True)=<xpra.common.KeyEvent object at 0x0000022d48640ee0>
23,384 client   1 @36.047 handle_key_action(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), <xpra.common.KeyEvent object at 0x0000022d48640ee0>) wid=1
23,384 client   1 @36.047 key_handled_as_shortcut(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), 'U+001B', ['control', 'mod2'], True) shortcuts_enabled=True, shortcuts=None
23,385 client   1 @36.047 send_delayed_key() delayed_event=None
23,385 client   1 @36.063 send_key_action(1, <xpra.common.KeyEvent object at 0x0000022d48640ee0>)
23,389 will try levels: [0, 4, 1, 5, 2, 6, 3, 7]
23,389 do_get_keycode(219, 'U+001B', True, ['control', 'mod2'], 16777243, '\x1b', 0)=-1, 0 (keyname translation)
23,389 process_key_action(['key-action', 1, 'U+001B', True, ('control', 'mod2'), 16777243, '\x1b', 219, 0]) server keycode=-1, group=0
23,390 filtered_modifiers_set(['control', 'mod2'])={'mod2', 'control'}
23,390 filtered_modifiers_set(['control', 'mod2'])={'mod2', 'control'}
23,474 client   1 @36.140 mask_to_names(<flags GDK_CONTROL_MASK of type Gdk.ModifierType>) GetKeyState(VK_NUMLOCK)=1, names=['control', 'mod2']
23,475 client   1 @36.140 parse_key_event(<Gdk.EventKey object at 0x0000022d4847f310 (void at 0x0000022d48b17000)>, False)=<xpra.common.KeyEvent object at 0x0000022d49cf50a0>
23,475 client   1 @36.140 handle_key_action(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), <xpra.common.KeyEvent object at 0x0000022d49cf50a0>) wid=1
23,476 client   1 @36.140 key_handled_as_shortcut(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), 'U+001B', ['control', 'mod2'], False) shortcuts_enabled=True, shortcuts=None
23,476 client   1 @36.140 send_delayed_key() delayed_event=None
23,476 client   1 @36.140 send_key_action(1, <xpra.common.KeyEvent object at 0x0000022d49cf50a0>)
23,477 process_key_action(['key-action', 1, 'U+001B', False, ('control', 'mod2'), 16777243, '\x1b', 219, 0]) server keycode=34, group=0
23,477 set_keyboard_layout_group(0) ignored, value unchanged
23,477 filtered_modifiers_set(['control', 'mod2'])={'mod2', 'control'}
23,477 filtered_modifiers_set(['control', 'mod2'])={'mod2', 'control'}
23,477 is_modifier(34) not found
23,477 handle_key((1, False, 'U+001B', 16777243, 34, ['control', 'mod2'], False, True))
23,477 handle keycode 34: key U+001B was already unpressed, ignoring
23,581 client   1 @36.234 mask_to_names(<flags 0 of type Gdk.ModifierType>) GetKeyState(VK_NUMLOCK)=1, names=['mod2']
23,581 client   1 @36.234 parse_key_event(<Gdk.EventKey object at 0x0000022d4847fdb0 (void at 0x0000022d48b17000)>, False)=<xpra.common.KeyEvent object at 0x0000022d48640ee0>
23,581 client   1 @36.234 handle_key_action(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV420P)), <xpra.common.KeyEvent object at 0x0000022d48640ee0>) wid=1
23,581 client   1 @36.234 key_handled_as_shortcut(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV420P)), 'Control_L', ['mod2'], False) shortcuts_enabled=True, shortcuts=None
23,581 client   1 @36.234 process_key_event: Control_L pressed=False, with GetKeyState(VK_RMENU)=1
23,581 client   1 @36.234 send_delayed_key() delayed_event=None
23,581 client   1 @36.234 send_key_action(1, <xpra.common.KeyEvent object at 0x0000022d48640ee0>)
23,581 process_key_action(['key-action', 1, 'Control_L', False, ('mod2',), 65507, '', 17, 0]) server keycode=37, group=0
23,581 set_keyboard_layout_group(0) ignored, value unchanged
23,581 modifier 'control' ignored (in ignored keynames={'Control_R', 'Control_L'})
23,582 filtered_modifiers_set(['control', 'mod2'])={'mod2'}
23,582 filtered_modifiers_set(['mod2'])={'mod2'}
23,582 handle_key((1, False, 'Control_L', 65507, 37, ['mod2'], True, True))
23,582 handle keycode unpressing  37: key 'Control_L'
23,582 fake_key(37, False)
25,216 New unix-domain connection received
25,216  on '/run/user/1000/xpra/0/socket'
25,217 disabled debugging for:
25,218  - Logger(xpra.x11.x11_server_core, x11, server, keyboard)
25,218  - Logger(xpra.server.rfb.rfb_server, rfb, keyboard)
25,218  - Logger(importlib._bootstrap, x11, bindings, keyboard)
25,218  - Logger(xpra.x11.server_keyboard_config, keyboard)
25,218  - Logger(xpra.server.source.input_mixin, keyboard)
25,218  - Logger(xpra.gtk_common.keymap, keyboard)
25,218  - Logger(xpra.server.mixins.input_server, keyboard)
25,218  - Logger(xpra.x11.xkbhelper, x11, keyboard)
25,419 New unix-domain connection received
25,420  on '/run/user/1000/xpra/0/socket'
25,510 client   1 @38.188 disabled debugging for:
25,511 client   1 @38.188  - Logger(xpra.platform.win32.keyboard, keyboard)
25,511 client   1 @38.188  - Logger(xpra.client.gtk_base.gtk_client_window_base, keyboard, events)
25,512 client   1 @38.188  - Logger(xpra.platform.win32.gui, win32, keyboard)
25,512 client   1 @38.188  - Logger(xpra.client.keyboard_helper, keyboard)
25,513 client   1 @38.188  - Logger(xpra.client.ui_client_base, client, keyboard)
25,513 client   1 @38.188  - Logger(xpra.client.keyboard_shortcuts_parser, keyboard)
25,514 client   1 @38.188  - Logger(xpra.client.gtk_base.gtk_client_window_base, keyboard)
25,514 client   1 @38.188  - Logger(xpra.gtk_common.keymap, keyboard)
25,514 client   1 @38.188  - Logger(xpra.client.client_window_base, keyboard)
25,515 client   1 @38.188  - Logger(xpra.platform.keyboard_base, keyboard)
```
Client:
```py
51,456 enabled debugging for:
51,457  - Logger(xpra.platform.win32.keyboard, keyboard)
51,457  - Logger(xpra.client.gtk_base.gtk_client_window_base, keyboard, events)
51,458  - Logger(xpra.platform.win32.gui, win32, keyboard)
51,458  - Logger(xpra.client.keyboard_helper, keyboard)
51,458  - Logger(xpra.client.ui_client_base, client, keyboard)
51,458  - Logger(xpra.client.keyboard_shortcuts_parser, keyboard)
51,459  - Logger(xpra.client.gtk_base.gtk_client_window_base, keyboard)
51,459  - Logger(xpra.gtk_common.keymap, keyboard)
51,459  - Logger(xpra.client.client_window_base, keyboard)
51,460  - Logger(xpra.platform.keyboard_base, keyboard)
52,822 mask_to_names(<flags GDK_CONTROL_MASK of type Gdk.ModifierType>) GetKeyState(VK_NUMLOCK)=1, names=['control', 'mod2']
52,823 parse_key_event(<Gdk.EventKey object at 0x0000022d485f3680 (void at 0x0000022d48b17000)>, True)=<xpra.common.KeyEvent object at 0x0000022d48640e20>
52,823 handle_key_action(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), <xpra.common.KeyEvent object at 0x0000022d48640e20>) wid=1
52,824 key_handled_as_shortcut(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), 'Control_R', ['control', 'mod2'], True) shortcuts_enabled=True, shortcuts=None
52,824 send_delayed_key() delayed_event=None
52,824 send_key_action(1, <xpra.common.KeyEvent object at 0x0000022d48640e20>)
52,943 mask_to_names(<flags GDK_CONTROL_MASK of type Gdk.ModifierType>) GetKeyState(VK_NUMLOCK)=1, names=['control', 'mod2']
52,944 parse_key_event(<Gdk.EventKey object at 0x0000022d485f3680 (void at 0x0000022d48b17000)>, True)=<xpra.common.KeyEvent object at 0x0000022d49cf50a0>
52,944 handle_key_action(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), <xpra.common.KeyEvent object at 0x0000022d49cf50a0>) wid=1
52,945 key_handled_as_shortcut(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), 'U+001B', ['control', 'mod2'], True) shortcuts_enabled=True, shortcuts=None
52,945 send_delayed_key() delayed_event=None
52,945 send_key_action(1, <xpra.common.KeyEvent object at 0x0000022d49cf50a0>)
53,050 mask_to_names(<flags 0 of type Gdk.ModifierType>) GetKeyState(VK_NUMLOCK)=1, names=['mod2']
53,050 parse_key_event(<Gdk.EventKey object at 0x0000022d485f32c0 (void at 0x0000022d48b17160)>, False)=<xpra.common.KeyEvent object at 0x0000022d48640e20>
53,051 handle_key_action(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV422P)), <xpra.common.KeyEvent object at 0x0000022d48640e20>) wid=1
53,051 key_handled_as_shortcut(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV422P)), 'Control_R', ['mod2'], False) shortcuts_enabled=True, shortcuts=None
53,052 send_delayed_key() delayed_event=None
53,052 send_key_action(1, <xpra.common.KeyEvent object at 0x0000022d48640e20>)
53,066 mask_to_names(<flags 0 of type Gdk.ModifierType>) GetKeyState(VK_NUMLOCK)=1, names=['mod2']
53,066 parse_key_event(<Gdk.EventKey object at 0x0000022d485f3a90 (void at 0x0000022d48b17000)>, False)=<xpra.common.KeyEvent object at 0x0000022d48640ee0>
53,066 handle_key_action(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV420P)), <xpra.common.KeyEvent object at 0x0000022d48640ee0>) wid=1
53,067 key_handled_as_shortcut(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV420P)), 'bracketleft', ['mod2'], False) shortcuts_enabled=True, shortcuts=None
53,067 send_delayed_key() delayed_event=None
53,067 send_key_action(1, <xpra.common.KeyEvent object at 0x0000022d48640ee0>)
54,673 mask_to_names(<flags GDK_CONTROL_MASK of type Gdk.ModifierType>) GetKeyState(VK_NUMLOCK)=1, names=['control', 'mod2']
54,674 parse_key_event(<Gdk.EventKey object at 0x0000022d4a899c20 (void at 0x0000022d48b17000)>, True)=<xpra.common.KeyEvent object at 0x0000022d48640ee0>
54,675 handle_key_action(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), <xpra.common.KeyEvent object at 0x0000022d48640ee0>) wid=1
54,676 key_handled_as_shortcut(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), 'Control_L', ['control', 'mod2'], True) shortcuts_enabled=True, shortcuts=None
54,676 process_key_event: Control_L pressed=True, with GetKeyState(VK_RMENU)=1
54,728 send_delayed_key() delayed_event=(<bound method KeyboardHelper.send_key_action of <xpra.client.gtk_base.gtk_keyboard_helper.GTKKeyboardHelper object at 0x0000022d68823460>>, 1, <xpra.common.KeyEvent object at 0x0000022d48640ee0>)
54,728 send_delayed_key() GetKeyState(VK_RMENU)=1
54,729 send_key_action(1, <xpra.common.KeyEvent object at 0x0000022d48640ee0>)
54,818 mask_to_names(<flags GDK_CONTROL_MASK of type Gdk.ModifierType>) GetKeyState(VK_NUMLOCK)=1, names=['control', 'mod2']
54,819 parse_key_event(<Gdk.EventKey object at 0x0000022d4847fdb0 (void at 0x0000022d48b17160)>, True)=<xpra.common.KeyEvent object at 0x0000022d48640ee0>
54,820 handle_key_action(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), <xpra.common.KeyEvent object at 0x0000022d48640ee0>) wid=1
54,820 key_handled_as_shortcut(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), 'U+001B', ['control', 'mod2'], True) shortcuts_enabled=True, shortcuts=None
54,821 send_delayed_key() delayed_event=None
54,821 send_key_action(1, <xpra.common.KeyEvent object at 0x0000022d48640ee0>)
54,911 mask_to_names(<flags GDK_CONTROL_MASK of type Gdk.ModifierType>) GetKeyState(VK_NUMLOCK)=1, names=['control', 'mod2']
54,911 parse_key_event(<Gdk.EventKey object at 0x0000022d4847f310 (void at 0x0000022d48b17000)>, False)=<xpra.common.KeyEvent object at 0x0000022d49cf50a0>
54,912 handle_key_action(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), <xpra.common.KeyEvent object at 0x0000022d49cf50a0>) wid=1
54,912 key_handled_as_shortcut(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV444P)), 'U+001B', ['control', 'mod2'], False) shortcuts_enabled=True, shortcuts=None
54,913 send_delayed_key() delayed_event=None
54,913 send_key_action(1, <xpra.common.KeyEvent object at 0x0000022d49cf50a0>)
54,996 mask_to_names(<flags 0 of type Gdk.ModifierType>) GetKeyState(VK_NUMLOCK)=1, names=['mod2']
54,997 parse_key_event(<Gdk.EventKey object at 0x0000022d4847fdb0 (void at 0x0000022d48b17000)>, False)=<xpra.common.KeyEvent object at 0x0000022d48640ee0>
54,998 handle_key_action(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV420P)), <xpra.common.KeyEvent object at 0x0000022d48640ee0>) wid=1
54,998 key_handled_as_shortcut(GLClientWindow(1 : GLDrawingArea(1, (1920, 1080), YUV420P)), 'Control_L', ['mod2'], False) shortcuts_enabled=True, shortcuts=None
54,998 process_key_event: Control_L pressed=False, with GetKeyState(VK_RMENU)=1
54,999 send_delayed_key() delayed_event=None
54,999 send_key_action(1, <xpra.common.KeyEvent object at 0x0000022d48640ee0>)
56,947 disabled debugging for:
56,948  - Logger(xpra.platform.win32.keyboard, keyboard)
56,948  - Logger(xpra.client.gtk_base.gtk_client_window_base, keyboard, events)
56,949  - Logger(xpra.platform.win32.gui, win32, keyboard)
56,949  - Logger(xpra.client.keyboard_helper, keyboard)
56,950  - Logger(xpra.client.ui_client_base, client, keyboard)
56,950  - Logger(xpra.client.keyboard_shortcuts_parser, keyboard)
56,950  - Logger(xpra.client.gtk_base.gtk_client_window_base, keyboard)
56,950  - Logger(xpra.gtk_common.keymap, keyboard)
56,951  - Logger(xpra.client.client_window_base, keyboard)
56,952  - Logger(xpra.platform.keyboard_base, keyboard)
```

**System Information (please complete the following information):**
- Server OS: Ubuntu 20.04.3 LTS
- Client OS: Microsoft Windows [Version 10.0.18363.836]
- Xpra Server Version (see above)
- Xpra Client Version (see above)

![image](https://user-images.githubusercontent.com/133706/148647962-487c6fc8-b34a-4a29-bfdc-7c4291308d05.png)

**Comment by totaam** [comment]
We can't find the key that MS Windows is giving us:
```
do_get_keycode(219, 'U+001B', True, ['control', 'mod2'], 16777243, '\x1b', 0)=-1, 0 (keyname translation)
```
That's because `U+001B` is not a valid keyname, it is kind of `Escape` but not (..)
I see the same think when running the keyboard tool from the `xpra_cmd.exe toolbox` on MS Windows.

We do see `brackleft` but only after `Control` has been unpressed:
```
process_key_action(['key-action', 1, 'bracketleft', False, ('mod2',), 91, '[', 219, 0]) server keycode=34, group=0
```

The commit above is really nasty but it does seem to work!
Surely there is a better way of doing this key mapping, but I don't see it.

@stdedos Please close if this works for you. (new beta build uploaded).

**Comment by stdedos** [comment]
This is now fixed:

```
2022-01-11 13:01:50,299 Xpra GTK3 client version 4.4-r30805 (g9c648c104) 64-bit
2022-01-11 13:01:50,316  running on Microsoft Windows 10
2022-01-11 13:01:51,946 GStreamer version 1.18.5 for Python 3.9.7 64-bit
2022-01-11 13:01:52,039 created named pipe 'Xpra\27864'
2022-01-11 13:01:52,072 keyboard layout code 0x408
2022-01-11 13:01:52,073 identified as 'Greek' : gr
2022-01-11 13:01:52,106 OpenGL_accelerate module loaded
2022-01-11 13:01:52,123 Using accelerated ArrayDatatype
2022-01-11 13:01:52,560 OpenGL accelerate missing: numpy_formathandler
2022-01-11 13:01:52,562 OpenGL enabled with NVIDIA GeForce RTX 2080 SUPER/PCIe/SSE2
2022-01-11 13:01:52,618  keyboard settings: layout=gr
2022-01-11 13:01:52,619  desktop size is 4480x1535:
2022-01-11 13:01:52,619   Default (1185x406 mm - DPI: 96x96) workarea: 4480x1495
2022-01-11 13:01:52,619     Samsung C32JG5x  2560x1440 at    0x0    (697x392 mm - DPI: 93x93) workarea: 2560x1400
2022-01-11 13:01:52,620     LG IPS234        1920x1080 at 2560x455  (510x290 mm - DPI: 96x95) workarea: 1920x1040 at 2560x455
2022-01-11 13:01:52,620  upscaled to 125%, virtual screen size: 3584x1228
2022-01-11 13:01:52,620   Default (1185x406 mm - DPI: 77x77) workarea: 3584x1196
2022-01-11 13:01:52,621     Samsung C32JG5x  2048x1152 at    0x0    (697x392 mm - DPI: 75x75) workarea: 2048x1120
2022-01-11 13:01:52,621     LG IPS234        1536x864  at 2048x364  (510x290 mm - DPI: 76x76) workarea: 1536x832  at 2048x364
2022-01-11 13:02:18,709 enabled remote logging
2022-01-11 13:02:18,709 Xpra shadow server version 4.4-r30771 64-bit
2022-01-11 13:02:18,710  running on Linux Ubuntu 20.04 focal
```

**Comment by stdedos** [comment]
... However, `Ctrl+]` does not work, similarly 😕

Do you want a new ticket, or you'll handle it "internally"?

**Comment by totaam** [comment]
Oops. Sorry. Ahem, that was obviously just a trial run, the more complete fix is in the next commit with the `XPRA_UNICODE_KEYNAMES=1` option to disable it and revert to the old behaviour.
(builds r30811 or later)

This will catch any key showing up as `U+NNNN` and try to find an actual keyname to use instead.

**Comment by stdedos** [comment]
> Ahem, that was obviously just a trial run

Ooof course, of course - that's how software is developed 🧐 
(I hope you appreciate my poor attempt at a monocle-guy kind of humor 😛)
